### PR TITLE
api:tenant default OAuth scope, API updates

### PR DESCRIPTION
**Pull request by tallen-bluecanvas**
* Adds call for archive endpoint.
* Adds `sourceCommit`, `mergeBranchName` to event payload.
* Removes unsupported `name` from Check request body.
* Removes verbose request/response logging.
* Ignores IntelliJ settings `.idea`.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -84,16 +87,11 @@ export class Client {
    * Creates an Axios client with default options and logging attached.
    * @internal
    */
-  private createAxios(config?: AxiosRequestConfig, disableLogging?: boolean): AxiosInstance {
+  private createAxios(config?: AxiosRequestConfig): AxiosInstance {
     const instance = axios.create(config);
     instance.defaults.headers = this.options.extraHeaders || {};
-    if (disableLogging) {
```
Comment (xoob) [inline src/client.ts:90]: Without this condition, the request and response body of the OAuth 2.0 credentials exchange are logged. That’s a security issue we must defend against. (https://cwe.mitre.org/data/definitions/532.html)

File: src/client.ts [hunk 2]
```diff
@@ -166,3 +164,28 @@ class DeploymentsClient {
...
+  constructor(axios: AxiosInstance) {
+    this.axios = axios;
+  }
+
+  /**
+   * Fetches a repository snapshot for the specified git revision as a gzipped tarball.
+   *
+   * @see https://docs.bluecanvas.io/reference/checks-api#get-archive
+   */
+  async getTarGzipBlob({ revision }: ArchivesGetTarGzipBlobRequest): Promise<ArchivesGetTarGzipBlobResponse> {
```
Comment (xoob) [inline src/client.ts]: Will I be able to fetch the `deploymentBranchName` ref from this API? (That would be needed IMO to make it work with DR files in the PMD use case, where we want to process the prepared files with overrides.) in that case, the name `ref` would be more accurate.

File: src/types/notifications.ts [hunk 3]
```diff
@@ -17,7 +17,9 @@ interface Deployment {
   phase: DeploymentPhase;
   state: DeploymentState;
   sourceBranchName: string;
+  sourceCommit: string;
   targetBranchName: string;
+  mergeBranchName: string;
```
Comment (xoob) [inline src/types/notifications.ts]: I suggest to call this `deploymentBranchName` on the public API, for clarity.

**Review by xoob — CHANGES_REQUESTED** [review]

File: src/client.ts (hunk 1 above)
Comment (tallen-bluecanvas) [reply src/client.ts:90]: Good catch, intended change in 9848f05c0249e32457efcfa0cafc23f98a124dc0

File: src/client.ts (hunk 2 above)
Comment (tallen-bluecanvas) [reply src/client.ts]: Yes the API supports branch refs, updated the name to `ref` in b9f1637ccefc13c9f70af1fba9f77455e0da8c48

File: src/types/notifications.ts (hunk 3 above)
Comment (tallen-bluecanvas) [reply src/types/notifications.ts]: Good idea, updated in c07067fad6f8727637f5fcf88c16d9d023bfced1

**Review by xoob — APPROVED** [review]
Looks good now!

**Pull request merged**